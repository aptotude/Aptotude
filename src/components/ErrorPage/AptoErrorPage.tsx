import React from 'react';
import './aptoErrorPage.scss';
import { AptoRow } from '../Row';
import { AptoCol } from '../Column';
import classNames from 'classnames';
import { AptoHeader } from '../Header';
import { AptoParagraph } from '../Paragraph';

interface AptoErrorPageProps {
  type?: '500' | '400' | '404' | '403';
  message?: string;
  title?: string;
}

const COMPONENT_PREFIX = 'AptoErrorPage';

export const AptoErrorPage = (props: AptoErrorPageProps) => {
  const { type, message, title } = props;

  const classes = classNames(
    COMPONENT_PREFIX,
    type && `${COMPONENT_PREFIX}--${type}`
  );

  let pageMessage = '';
  let pageTitle = '';

  if (type === '400') {
    pageTitle = 'Bad Request';
    pageMessage = 'Sorry, there was something ';
  } else if (type === '404') {
    pageTitle = 'Page Not Found';
    pageMessage = 'Sorry, but the page you were trying to view does not exist.';
  } else if (type === '403') {
    pageTitle = 'Unauthorized';
    pageMessage = 'Sorry, you do not have permission to view this page.';
  } else if (type === '500') {
    pageTitle = 'Internal Server Error';
    pageMessage = 'An internal server error occurred.';
  } else {
    pageTitle = 'Server Error';
  }

  if (message) {
    pageMessage = message;
  }

  if (title) {
    pageTitle = title;
  }

  return (
    <AptoRow align="center" className={classes}>
      <AptoCol>
        <AptoHeader type="2" transform={false}>
          {pageTitle}
        </AptoHeader>
        {pageMessage && <AptoParagraph>{pageMessage}</AptoParagraph>}
      </AptoCol>
    </AptoRow>
  );
};
