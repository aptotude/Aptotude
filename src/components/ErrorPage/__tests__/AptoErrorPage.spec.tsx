import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoErrorPage } from '../AptoErrorPage';

afterEach(() => {
  cleanup();
});

describe('AptoErrorPage', () => {
  it('Renders Component', () => {
    const { container } = render(<AptoErrorPage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders 500', () => {
    const { container } = render(<AptoErrorPage type="500" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders 400', () => {
    const { container } = render(<AptoErrorPage type="400" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders 404', () => {
    const { container } = render(<AptoErrorPage type="404" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders 403', () => {
    const { container } = render(<AptoErrorPage type="403" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders message', () => {
    const { container } = render(<AptoErrorPage message="cat is fat" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders title', () => {
    const { container } = render(<AptoErrorPage title="hi title" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
