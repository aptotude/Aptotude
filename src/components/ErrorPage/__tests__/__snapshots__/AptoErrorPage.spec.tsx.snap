// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoErrorPage Renders 400 1`] = `
<div
  class="AptoRow AptoRow--alignCenter AptoErrorPage AptoErrorPage--400"
>
  <div
    class="AptoCol"
  >
    <h2
      class="AptoHeader AptoHeader--h2 AptoHeader--noTransform"
    >
      Bad Request
    </h2>
    <p
      class="AptoParagraph"
    >
      Sorry, there was something 
    </p>
  </div>
</div>
`;

exports[`AptoErrorPage Renders 403 1`] = `
<div
  class="AptoRow AptoRow--alignCenter AptoErrorPage AptoErrorPage--403"
>
  <div
    class="AptoCol"
  >
    <h2
      class="AptoHeader AptoHeader--h2 AptoHeader--noTransform"
    >
      Unauthorized
    </h2>
    <p
      class="AptoParagraph"
    >
      Sorry, you do not have permission to view this page.
    </p>
  </div>
</div>
`;

exports[`AptoErrorPage Renders 404 1`] = `
<div
  class="AptoRow AptoRow--alignCenter AptoErrorPage AptoErrorPage--404"
>
  <div
    class="AptoCol"
  >
    <h2
      class="AptoHeader AptoHeader--h2 AptoHeader--noTransform"
    >
      Page Not Found
    </h2>
    <p
      class="AptoParagraph"
    >
      Sorry, but the page you were trying to view does not exist.
    </p>
  </div>
</div>
`;

exports[`AptoErrorPage Renders 500 1`] = `
<div
  class="AptoRow AptoRow--alignCenter AptoErrorPage AptoErrorPage--500"
>
  <div
    class="AptoCol"
  >
    <h2
      class="AptoHeader AptoHeader--h2 AptoHeader--noTransform"
    >
      Internal Server Error
    </h2>
    <p
      class="AptoParagraph"
    >
      An internal server error occurred.
    </p>
  </div>
</div>
`;

exports[`AptoErrorPage Renders Component 1`] = `
<div
  class="AptoRow AptoRow--alignCenter AptoErrorPage"
>
  <div
    class="AptoCol"
  >
    <h2
      class="AptoHeader AptoHeader--h2 AptoHeader--noTransform"
    >
      Server Error
    </h2>
    
  </div>
</div>
`;

exports[`AptoErrorPage Renders message 1`] = `
<div
  class="AptoRow AptoRow--alignCenter AptoErrorPage"
>
  <div
    class="AptoCol"
  >
    <h2
      class="AptoHeader AptoHeader--h2 AptoHeader--noTransform"
    >
      Server Error
    </h2>
    <p
      class="AptoParagraph"
    >
      cat is fat
    </p>
  </div>
</div>
`;

exports[`AptoErrorPage Renders title 1`] = `
<div
  class="AptoRow AptoRow--alignCenter AptoErrorPage"
>
  <div
    class="AptoCol"
  >
    <h2
      class="AptoHeader AptoHeader--h2 AptoHeader--noTransform"
    >
      hi title
    </h2>
    
  </div>
</div>
`;
